### PR TITLE
add a List() function to pcstore to list all pod clusters

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/kp/pcstore/consul_store_test.go [hunk 1]
```diff
@@ -231,6 +231,57 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestList(t *testing.T) {
+	store := consulStoreWithFakeKV()
+	podID := types.PodID("pod_id")
+	az := fields.AvailabilityZone("us-west")
+	clusterName := fields.ClusterName("cluster_name")
+
+	selector := klabels.Everything().
```
Comment (rudle) [inline pkg/kp/pcstore/consul_store_test.go]: You probably already know there's a helper for this, FindWhereLabeled()

**Comment by rudle** [comment]
:ship:

File: pkg/kp/pcstore/consul_store_test.go (hunk 1 above)
Comment (mpuncel) [inline pkg/kp/pcstore/consul_store_test.go]: I don't actually care about the content of the selector here so i'll make it `klabels.Everything()`

**Comment by mpuncel** [comment]
made the test use a klabels.Everything() to keep it more concise and added List() to fake pcstore

**Comment by rudle** [comment]
:ship: :ship:

**Pull request merged**